planner/core/rule_join_elimination.go

```diff
@@ -201,22 +201,21 @@ func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Co
 
 	switch x := p.(type) {
 	case *LogicalProjection:
-		parentCols = make([]*expression.Column, 0, len(x.Exprs))
+		parentCols = parentCols[:0]
 		for _, expr := range x.Exprs {
 			parentCols = append(parentCols, expression.ExtractColumns(expr)...)
 		}
 	case *LogicalAggregation:
-		parentCols = x.groupByCols
+		copy(parentCols, x.groupByCols)
```

Comment:
A better way is to use the following form:
```go
parentCols = append(parentCols[:0], x.groupByCols...)
```

The difference is, if `len(parentCols)` is smaller than `len(x.groupByCols)`, it will only copy `len(parentCols)` columns, which is incorrect. You can verify this behavior from the documentation of `copy()` function: https://golang.org/pkg/builtin/#copy